internal/trigger/http/rest_test.go

```diff
@@ -138,6 +140,88 @@ func TestTriggerRequestHandler_ProcessError(t *testing.T) {
+	data := []byte("some data")
+	contentType := "arbitrary string"
+	correlationId := uuid.NewString()
+	afc := &interfaceMocks.AppFunctionContext{}
+
+	bnd := &triggerMocks.ServiceBinding{}
+	bnd.On("LoggingClient").Return(logger.NewMockClient())
+	bnd.On("BuildContext", mock.AnythingOfType("types.MessageEnvelope")).Return(afc)
+	bnd.On("RawInput").Return(false)
+
+	mp := &triggerMocks.MessageProcessor{}
+	mp.On("MessageReceived", afc, mock.AnythingOfType("types.MessageEnvelope"), mock.AnythingOfType("interfaces.PipelineResponseHandler")).Return(func(ctx interfaces.AppFunctionContext, env types.MessageEnvelope, f interfaces.PipelineResponseHandler) error {
+		assert.Equal(t, correlationId, env.CorrelationID)
+		assert.Equal(t, afc, ctx)
+		assert.Equal(t, data, env.Payload)
+		assert.Equal(t, contentType, env.ContentType)
+		return errors.NewCommonEdgeX(errors.KindContractInvalid, "error", nil)
+	})
+
+	trigger := Trigger{
+		serviceBinding:   bnd,
+		messageProcessor: mp,
+	}
+
+	writer := &mocks.TriggerResponseWriter{}
+	writer.On("WriteHeader", http.StatusBadRequest)
+	writer.On("Write", []byte("error")).Return(0, nil)
+
+	req, err := http.NewRequest("", "", bytes.NewBuffer(data))
+	req.Header = http.Header{}
+	req.Header.Add(common.ContentType, contentType)
+	req.Header.Add(common.CorrelationHeader, correlationId)
+
+	require.NoError(t, err)
+
+	trigger.requestHandler(writer, req)
+
+	writer.AssertExpectations(t)
+}
+
+func TestTriggerRequestHandler_ProcessError_ContractInvalid_RawInput(t *testing.T) {
+	data := []byte("some data")
+	contentType := "arbitrary string"
+	correlationId := uuid.NewString()
+	afc := &interfaceMocks.AppFunctionContext{}
+
+	bnd := &triggerMocks.ServiceBinding{}
+	bnd.On("LoggingClient").Return(logger.NewMockClient())
+	bnd.On("BuildContext", mock.AnythingOfType("types.MessageEnvelope")).Return(afc)
+	bnd.On("RawInput").Return(true)
+
+	mp := &triggerMocks.MessageProcessor{}
+	mp.On("MessageReceived", afc, mock.AnythingOfType("types.MessageEnvelope"), mock.AnythingOfType("interfaces.PipelineResponseHandler")).Return(func(ctx interfaces.AppFunctionContext, env types.MessageEnvelope, f interfaces.PipelineResponseHandler) error {
+		assert.Equal(t, correlationId, env.CorrelationID)
+		assert.Equal(t, afc, ctx)
+		assert.Equal(t, data, env.Payload)
+		assert.Equal(t, contentType, env.ContentType)
+		return errors.NewCommonEdgeX(errors.KindContractInvalid, "error", nil)
+	})
+
+	trigger := Trigger{
+		serviceBinding:   bnd,
+		messageProcessor: mp,
+	}
+
+	writer := &mocks.TriggerResponseWriter{}
+	writer.On("WriteHeader", http.StatusUnprocessableEntity)
+	writer.On("Write", []byte("error")).Return(0, nil)
+
+	req, err := http.NewRequest("", "", bytes.NewBuffer(data))
+	req.Header = http.Header{}
+	req.Header.Add(common.ContentType, contentType)
+	req.Header.Add(common.CorrelationHeader, correlationId)
+
+	require.NoError(t, err)
+
+	trigger.requestHandler(writer, req)
+
+	writer.AssertExpectations(t)
```

Comment:
What asserts is this running?
I haven't used the mocks in this way to validate the test....

Reply:
It validates that the void setups on write - so if I set up `WriteHeader` above with `BadRequest` but `UnprocessableEntity` was passed the test will fail.

Reply:
I thought that the mocks would panic if you it received a call that didn't match one on the .On() statements. i.e. not `BadRequest`

Reply:
But I assume the AssertExpectations checks that it was actually called...